zerver/actions/message_flags.py

```diff
@@ -288,6 +290,13 @@ def do_update_message_flags(
                 if message_id not in message_ids_in_unsubscribed_streams
             ]
 
+            skipped_marking_unread_stream_ids = list(
+                {
+                    access_message(user_profile, message_id)[0].recipient.type_id
```

Comment:
I was using `Message.objects.get(id=message_id)` to retrieve the message object here, and the linter suggested using `access_message` to fetch the message object. However, after some testing, there is a bug in the code. If the narrow contains a message from an unsubscribed private stream, we will get an error because `access_message` does not permit accessing messages from unsubscribed private streams.

Please suggest on which query or function to use to get the message object in this case.

Reply:
https://chat.zulip.org/#narrow/stream/3-backend/topic/update.20has_message_access.3F/near/1638961